### PR TITLE
[10.0][IMP] compare_noupdate_xml_records: analyze repositories

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
This PR introduce a third optional argument, the mode, that is `module` or `repository`:
* The 'module' option does the usual analysis between two modules. In this case, the two first arguments are the module paths. Default option.
* The 'repository' option compares all the modules that have the same name in both repositories. In this case, the two first arguments are the repository paths.

### CODE REVIEW

File: scripts/compare_noupdate_xml_records.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 # Copyright 2016 Opener B.V. <https://opener.am>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
+from odoo.modules.module import MANIFEST_NAMES
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: This import is not valid.

File: scripts/compare_noupdate_xml_records.py [hunk 2]
```diff
@@ -211,5 +182,76 @@ def main(argv=None):
...
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'olddir', metavar='older_directory')
+    parser.add_argument(
+        'newdir', metavar='newer_directory')
+    parser.add_argument(
+        'mode', metavar='module/repository')
+    arguments = parser.parse_args(argv)
+
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: Don't let these blank lines. They are IMO very anti-aesthetic and makes harder to read method lines together

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: scripts/compare_noupdate_xml_records.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply scripts/compare_noupdate_xml_records.py]: Deleted all blank lines 😅

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza, is it ok now? 🤓

**Comment by pedrobaeza** [comment]
I have to re-check. I'll tell you

File: scripts/compare_noupdate_xml_records.py [hunk 3]
```diff
@@ -211,5 +181,68 @@ def main(argv=None):
...
+    - Does not handle the shorthand records <menu>, <act_window> etc.,
+      although that could be done using the same expansion logic as
+      is used in their parsers in openerp/tools/convert.py
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'olddir', metavar='older_directory')
+    parser.add_argument(
+        'newdir', metavar='newer_directory')
+    parser.add_argument(
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py:216]: Can you make this argument optional and default to `module` for having the same command line as previously?

Comment (MiquelRForgeFlow) [reply scripts/compare_noupdate_xml_records.py:216]: Done :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**